File: common/src/main/java/com/skm/sftp/SFTPBlackBox.java
package com.skm.sftp;


import com.jcraft.jsch.JSchException;
import com.jcraft.jsch.SftpException;

import java.io.IOException;

public interface SFTPBlackBox {
  void upload(
      String remoteHost,
      String username,
      String password,
      String localPath,
      String remotePath) throws IOException, JSchException, SftpException;
  void download(
      String remoteHost,
      String username,
      String password,
      String localPath,
      String remotePath) throws IOException, JSchException, SftpException;
}

File: common/src/main/kotlin/dev/scottkey/sftp/SFTPBlackBox.kt
package dev.scottkey.sftp

interface SFTPBlackBox {
    fun upload(
        remoteHost: String,
        username: String,
        password: String,
        localPath: String,
        remotePath: String
    )

    fun download(
        remoteHost: String,
        username: String,
        password: String,
        localPath: String,
        remotePath: String
    )
}